### PR TITLE
Xcode9

### PR DESCRIPTION
**Pull request by godrei**

### CODE REVIEW

File: cmd/xcode.go [hunk 1]
```diff
@@ -25,130 +28,171 @@ var xcodeCmd = &cobra.Command{
...
-			fmt.Println()
-			fmt.Println()
-			log.Println("🔦  Scanning Schemes ...")
-			schemes, err := xcodeCmd.ScanSchemes()
-			if err != nil {
-				return printXcodeScanFinishedWithError("Failed to scan Schemes: %s", err)
-			}
-			log.Debugf("schemes: %v", schemes)
+		fmt.Println()
+		selectedScheme, err := goinp.SelectFromStringsWithDefault("Select the Scheme you usually use in Xcode", 1, schemes)
```
Comment (viktorbenei) [inline cmd/xcode.go]: if there's only one `schemes` then auto select that

File: osxkeychain/osxkeychain.go [hunk 2]
```diff
@@ -140,7 +139,7 @@ type IdentityWithRefModel struct {
 // FindAndValidateIdentity ...
 //  IMPORTANT: you have to C.CFRelease the returned items (one-by-one)!!
 //             you can use the ReleaseIdentityWithRefList method to do that
-func FindAndValidateIdentity(identityLabel string, isFullLabelMatch bool) ([]IdentityWithRefModel, error) {
+func FindAndValidateIdentity(identityLabel string, isFullLabelMatch bool) (*IdentityWithRefModel, error) {
 	foundIdentityRefs, err := FindIdentity(identityLabel, isFullLabelMatch)
```
Comment (viktorbenei) [inline osxkeychain/osxkeychain.go]: if you don't allow partial match / glob match anymore then don't accept a `isFullLabelMatch` param and call `FindIdentity` with `isFullLabelMatch = true` directly

File: xamarin/xamarin.go [hunk 3]
```diff
@@ -40,133 +35,109 @@ func (xamarinCmd *CommandModel) SetConfigurationPlatformCombination(configPlatfo
...
 		}
 	}
 
-	identities := []common.CodeSigningIdentityInfo{}
-	for _, v := range identitiesMap {
-		identities = append(identities, v)
+	if len(newArchives) == 0 {
+		return "", xamarinBuildOutput, fmt.Errorf("No archive generated during the build")
+	} else if len(newArchives) > 1 {
+		log.Warnf("Multiple archives generated during the build, using first ...")
```
Comment (viktorbenei) [inline xamarin/xamarin.go]: return an error instead

File: osxkeychain/osxkeychain.go [hunk 4]
```diff
@@ -212,14 +215,8 @@ func FindIdentity(identityLabel string, isFullLabelMatch bool) ([]IdentityWithRe
 			return nil, fmt.Errorf("FindIdentity: failed to get 'labl' property: %s", err)
```
Comment (trapacska) [inline osxkeychain/osxkeychain.go]: `labl` -> `label`

File: cmd/utils.go [hunk 5]
```diff
@@ -40,16 +40,16 @@ func exportMethod(group export.IosCodeSignGroup) string {
...
-			return &certificate
+			return certificate, nil
 		}
 		if strings.ToLower(certificate.SHA1Fingerprint) == strings.ToLower(nameOrSHA1Fingerprint) {
-			return &certificate
+			return certificate, nil
 		}
 	}
-	return nil
+	return certificateutil.CertificateInfoModel{}, fmt.Errorf("installed certificate not found with common name or sha1 hash: %s", nameOrSHA1Fingerprint)
```
Comment (viktorbenei) [inline cmd/utils.go]: use `errors` instead, the https://github.com/pkg/errors (`errors.Errorf`)

File: cmd/xcode.go [hunk 6]
```diff
@@ -90,13 +86,19 @@ the one you usually open in Xcode, then hit Enter.
...
+			return printXcodeScanFinishedWithError("No schemes found")
+		} else if len(schemes) == 1 {
+			schemeToUse = schemes[0]
+		} else {
+			fmt.Println()
+			selectedScheme, err := goinp.SelectFromStringsWithDefault("Select the Scheme you usually use in Xcode", 1, schemes)
+			if err != nil {
+				return printXcodeScanFinishedWithError("Failed to select Scheme: %s", err)
+			}
+			log.Debugf("selected scheme: %v", selectedScheme)
```
Comment (viktorbenei) [inline cmd/xcode.go]: I'd move this debug print out from this if, so that `if len(schemes) == 1 ` we'll still print the scheme, for debug purposes

**Pull request merged**